pkg/controllers/controlplanemachineset/controller.go

```diff
@@ -70,7 +70,7 @@ func (r *ControlPlaneMachineSetReconciler) SetupWithManager(mgr ctrl.Manager) er
 	if err := ctrl.NewControllerManagedBy(mgr).
 		For(&machinev1.ControlPlaneMachineSet{}, builder.WithPredicates(filterControlPlaneMachineSet(r.Namespace))).
 		Owns(&machinev1beta1.Machine{}, builder.WithPredicates(filterControlPlaneMachines(r.Namespace))).
-		Watches(&source.Kind{Type: &configv1.ClusterOperator{}}, handler.EnqueueRequestsFromMapFunc(clusterOperatorToControlPlaneMachineSet(r.Namespace, r.OperatorName))).
+		Watches(&source.Kind{Type: &configv1.ClusterOperator{}}, handler.EnqueueRequestsFromMapFunc(clusterOperatorToControlPlaneMachineSet(r.Namespace)), builder.WithPredicates(filterClusterOperator(r.OperatorName))).
```

Comment:
Might be worth a comment here to note that predicates are executed before the event handler is called

Also, this line got long, let's wrap this over a few lines
```suggestion
		Watches(
			&source.Kind{Type: &configv1.ClusterOperator{}},
									handler.EnqueueRequestsFromMapFunc(clusterOperatorToControlPlaneMachineSet(r.Namespace)),
			builder.WithPredicates(filterClusterOperator(r.OperatorName)),
		).
```